### PR TITLE
TST/STY: Hough transform regression test & Cython wrappers

### PR DESCRIPTION
**Pull request by JDWarner**
Adds a regression test for the behavior patched in #1421. 

Also, when testing this module I noted `hough_line`, `hough_ellipse`, and `probabilistic_hough_line` were missing function signatures because they were imported directly from Cython. These functions are now imported from pure Python wrappers. Along with this I moved some of the input checks into the new Python wrappers, which should make the generated C smaller and simpler.

Doctest imports now fixed in `hough_ellipse` - previously these doctests were not run, as they were exclusively in Cython.

### CODE REVIEW

File: skimage/transform/hough_transform.py [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+
+    .. plot:: hough_tf.py
+
+    """
+    if img.ndim != 2:
+        raise ValueError('The input image `img` must be 2D.')
+
+    if theta is None:
+        # These values are approximations of pi/2
+        theta = np.linspace(-1.5707963267948966, 1.5707963267948966, 180)
```
Comment (ahojnnes) [inline skimage/transform/hough_transform.py]: How about using np.pi for these constants?

**Comment by JDWarner** [comment]
@ahojnnes Done, but don't merge yet  - I'm going to go to 100% test coverage including the input checks from Cython.

**Comment by JDWarner** [comment]
And done, assuming the builds remain happy.

**Comment by ahojnnes** [comment]
Excellent, LGTM. Thanks Josh.

**Pull request merged**